libgloss/dramfs/sys_read.c

```diff
@@ -6,35 +6,40 @@
 /* Read from a file.  */
 ssize_t _read(int fd, void *ptr, size_t len)
 {
-  if(dramfs_check_fd(fd) < 0) {
-    return -1;
+  if (dramfs_check_fd(fd) < 0) {
```

Comment:
This seems like a semantic change. If check_fd < 0, we used to return -1. What's the difference in behavior?

Reply:
If no stdin was created in the lfs.c file(we're building as baremetal), check_fd is going to return -1, but we might want to read the host getchar.

Reply:
I'm think I'm missing something here. The MKLFS tool doesn't always define the STDIN/STDOUT to exist in lfs.c?

Reply:
MKLFS always will create an stdin in lfs.c. If we're using BAREMETAL perch then we're not using any lfs.c files so there's no stdin file descriptor.

Reply:
No exactly. If there is an lfs.c, we read the stdin file before doing the host_getchar. If not, we would only do host_getchar. Before this change, we only did host_getchar if there was a stdin file and we didn't actually read the file.

Reply:
Yeah I'm just talking about this if statement right now. If we don't call dramfs_init, will this go to host read or fail? Can we call dramfs_init without an LFS? I'm trying to nail down does this change fully enable users to use stdin/stdout without messing with lfs or will something down the line fail because of not having LFS or compiling with BAREMETAL

Reply:
If we don't call dramfs_init this will go to host_getchar. We cannot call the non-dummy dramfs_init without an lfs.c, for this reqason there's a default lfs.c with just the stdin/out/err in the repo. This if statement convers the BAREMETAL case. The other if statment after lfs_file_read covers the default lfs.c case.